src/Build/Evaluation/IntrinsicFunctions.cs

```diff
@@ -627,6 +628,12 @@ internal static bool AreFeaturesEnabled(Version wave)
             return ChangeWaves.AreFeaturesEnabled(wave);
         }
 
+        internal static string SubstringByTextElements(string input, int start, int length)
+        {
+            StringInfo stringInfo = new StringInfo(input);
+            return stringInfo.SubstringByTextElements(start, length);
+        }
```

Comment:
We've been adding intrinsics to the giant `TryExecuteWellKnownFunction` method in `Expander` to avoid the cost of reflection. It's not a huge deal but I think it would be better to add this one as well, especially since we know it's used in pretty much all builds.